### ISSUE TITLE
The secret infinite bank with random extra deposits

### ISSUE THREAD

**Issue by msouth** [ironbane/IronbaneServerLegacy#388]
(Image is from msouth  account's "Roara" character)

As with  #387 , I suppose this could be a feature I don't understand.  If I could get this to happen reliably, it _would_ be a feature, because I would get both extra storage space and occasional free stuff.

This is what it looks like when the glitch/feature is active:

![multi-bank](http://i40.tinypic.com/b97ssi.jpg)

I have seen one "sign" that you're about to get the infini-bank.  You try to put an item in the bank in what appears to be an open slot, and it won't accept it.  Then, you move back, move forward again, and you get the infini-bank.

When this "interface" is up, you can drag any of the items that you can see to any open inventory space you have.  I think once you do that the rest of the column might disappear...I can't remember for sure.

At the very bottom of this picture you can just make out the background that appears underneath when the stack of items is long enough that it goes under the game's normal display area--it's the "clear screen" background you see when ironbane is loading.  I believe as you mouse over the lower items the whole screen shifts up, and the lower items are displayed against this background.

I'm not sure that you always have to have the "non-functioning drop" thing happen first...I've definitely seen the infini-bank interface pop up right upon arrival, but that doesn't mean that I didn't have the non-droppable space happen on my previous visit and just forgot.

Also, once in a while you see things you never put in there.  I know that I never had blackthorn armor, for example, and I got some from the bottom of one of the stacks.

Coupled with  #387, this issue could be retitled:

"The bank giveth blackthorn, and the bank taketh away blackthorn".

:)

**Comment by ingmargoudt** [comment]
Yes, I have seen this 'infinite' bank behaviour before too, at least to the extent where you drop items under the bankslot. My assumption is some caching issues, but I have not looked into in it deeply.

**Comment by msouth** [comment]
> I have seen one "sign" that you're about to get the infini-bank. 
> You try to put an item in the bank in what appears to be an open slot, 
> and it won't accept it. Then, you move back, move forward again, and you get the infini-bank.

So, my n00blet account (one I'm using to re-start from the tutorial forward) got this glitch, as pictured:

![n00blet bank glitch](http://i41.tinypic.com/21mz0nr.png)

I believe I was attempting to drop something on the bank's position 9.  Position 9 appeared empty, and, I think, had never been populated, but the drop was unsuccessful.  Now that I know this is the omen for this glitch, I knew to immediately step back, let the interface disappear, and then step forward.  As I expected, I saw the glitch and position 9 was occupied with something I hadn't put there.

Since this is a very new character, I am 100% sure that I it had never had that wine thing that is showing in position 9, and I'm >90% sure that I had never put anything in that slot.

It's like the in-memory version has random data in there or something.

Anyway--I just wanted to update, because if someone is trying to reproduce it, I think putting stuff in there until you see a drop fail might save you a lot of time walking forward and back.

**Comment by msouth** [comment]
I thought #401 was dependent on this, but that turned out not to be the case.  In the process of trying to prove that, though, I might have come up with a formula for reproducing this one.

I went through the tutorial, doing the bow parkour, getting the hidden iron armor and doing the iron helm parkour as well.  (No reason, necessarily, to think those are part of the bug, I'm just reporting it all.)

Once I got out into the world, I dropped duplicate items and killed/collected until I had ten different items in inventory (again, may not be necessary, just doing it so I will know later which item was placed where, etc.).

Once I had ten distinct items, went to /stuck and then to the bank, and started loading the items into the bank with item 1 in bank's slot 1, item 2 in bank's slot 2, etc.

The drop failed for slot 2 (on the previous character it was slot 1), leaving me like this:

![right after item 2 failed to drop](http://i44.tinypic.com/2sah6bl.jpg)

 I stepped back to make the bank interface disappear.  Then I stepped forward to see this bug manifest (as expected, since the failed drop seems to be an omen for it)

![look at all that cheese](http://i43.tinypic.com/212ebnd.jpg)

Both of the characters I did this with are in #401 state, with their banks totally gone, so they are probably not good investigation candidates now.

**Comment by msouth** [comment]
Three times in a row I have followed the above procedure, and every time I got the bug (my most recent one was a slight variation on the bug, but I think it's the same thing.  I think it's safe to call this reproducible.

I've left TessSting in the msouth account in this state.  Coming back in through a new incognito window showed the mystery item (in this case a money bag with one coin) still there (the dull sword was successfully dropped into slot 1):

![bow would not drop, money bag from nowhere](http://i41.tinypic.com/16kqecm.jpg)

This was the same procedure as the previous two:

1) new character
2) go through tutorial, do all parkour including iron armor and helmet
3) killed big rat
4) in the world, killed a snowman and used the dropped snowball to replace my one duplicate item
5)  /stuck, go to bank
6) start dragging the items up to the bank one at a time until one fails to drop
7) As soon as you see the drop failure, step back until the bank interface is gone
8) Step forward again.  You should see an item in the bank slot that you were unable to drop your item on.

It might be worth looking at the bank structure in memory as soon as you walk up, before attempting to drop anything.  I suspect it will already have one or more things that it thinks are in the bank.

I am leaving TessSting in the condition above in case anyone wants to look at her data specifically.

**Comment by msouth** [comment]
I found a way to avoid the glitch when you see the omen.

go to bank
bank is empty
put milk in slot 1--success
put dull sword in slot 2--fails, dull sword stays in my inventory
[which items doesn't matter--swap milk with dull sword, same result]
[I think which slot doesn't matter--if it's going to happen, it happens on the second]
************_prevent glitch ***_******************
take milk back out--succeeds (and now you are in the same state as you were before)
step back
step forward
empty bank
*************\* I think taking the item back prevents the glitch from manifesting ******
To confirm that this prevented the glitch, I did it again without the prevent step, and the glitch showed:

put milk in slot 1--success
put dull sword in slot 2--fail
step back ****_without taking the milk back**_**
step forward, glitch shows
pull a knight helmet from the glitch "stack", add to inventory--success
attempt to pull a second item (brutal axe) from a different "stack"--fails
step back
step forward--empty bank

I went and got another cheap item, reproduced the bug, and successfully pulled the brutal axe.

I think since there was only one item in there "for real", even though I could choose anything from the items that shouldn't have been there, I could only choose one item.

Account/character:
msouth_bank:Sherpa

I had never used the bank for this character previously.
